x-pack/plugins/ml/public/jobs/jobs_list_new/components/jobs_list_view/jobs_list_view.js

```diff
@@ -0,0 +1,185 @@
+        itemIdToExpandedRowMap[jobId] = (
+          <JobDetails
+            jobId={jobId}
+            job={this.state.fullJobsList[jobId]}
+            addYourself={this.addUpdateFunction}
+          />
+        );
+      } else {
+        itemIdToExpandedRowMap[jobId] = (
+          <JobDetails
+            jobId={jobId}
+            addYourself={this.addUpdateFunction}
+          />
+        );
+      }
+
+      this.setState({ itemIdToExpandedRowMap }, () => {
+        loadFullJob(jobId)
+          .then((job) => {
+            const fullJobsList = { ...this.state.fullJobsList };
+            fullJobsList[jobId] = job;
+            this.setState({ fullJobsList }, () => {
+              // take a fresh copy of the itemIdToExpandedRowMap object
+              itemIdToExpandedRowMap = { ...this.state.itemIdToExpandedRowMap };
+              if (itemIdToExpandedRowMap[jobId] !== undefined) {
+                // wrap in a check, in case the user closes the expansion before the
+                // loading has finished
+                itemIdToExpandedRowMap[jobId] = (
+                  <JobDetails
+                    jobId={jobId}
+                    job={job}
+                    addYourself={this.addUpdateFunction}
+                    removeYourself={this.removeUpdateFunction}
+                  />
+                );
+              }
+              this.setState({ itemIdToExpandedRowMap });
+            });
+          })
+          .catch((error) => {
+            console.error(error);
+          });
+      });
+    }
+  }
+
+  addUpdateFunction = (id, f) => {
+    this.updateFunctions[id] = f;
+  }
+  removeUpdateFunction = (id) => {
+    delete this.updateFunctions[id];
+  }
+
+  setShowEditJobFlyoutFunction = (func) => {
+    this.showEditJobFlyout = func;
+  }
+  setShowDeleteJobModalFunction = (func) => {
+    this.showDeleteJobModal = func;
+  }
+  setShowStartDatafeedModalFunction = (func) => {
+    this.showStartDatafeedModal = func;
+  }
+
+  selectJobChange = (selectedJobs) => {
+    this.setState({ selectedJobs });
+  }
+
+  refreshJobSummaryList(autoRefresh = true) {
+    if (this.blockAutoRefresh === false) {
+      const expandedJobsIds = Object.keys(this.state.itemIdToExpandedRowMap);
+      ml.jobService.jobsSummary(expandedJobsIds)
+        .then((jobs) => {
+          const fullJobsList = {};
+          const jobsSummaryList = jobs.map((job) => {
+            if (job.fullJob !== undefined) {
+              fullJobsList[job.id] = job.fullJob;
+              delete job.fullJob;
+            }
+            return job;
+          });
+          this.setState({ jobsSummaryList, fullJobsList });
+
+          Object.keys(this.updateFunctions).forEach((j) => {
+            this.updateFunctions[j].setState({ job: fullJobsList[j] });
+          });
+
+          if (autoRefresh === true) {
+            setTimeout(() => {
+              this.refreshJobSummaryList();
+            }, 10000);
```

Comment:
Should this 10s refresh interval be defined as a constant?

Reply:
this is temporary, the plan is to allow the user to use kibana's auto-refresh feature.
i'd forgotten to add that to the meta issue.

Reply:
added to https://github.com/elastic/kibana/issues/20150